app/models/volunteer.rb

```diff
@@ -41,11 +46,11 @@ class Volunteer < ActiveRecord::Base
 
   # more trustworthy and self.assigned? attribute?
   def unassigned?
-    self.assignments.length == 0
+    assignments.empty?
   end
 
   def needs_training?
-    not self.logs.collect{ |l| l.complete }.any?
+    logs.none?(&:complete)
```

Comment:
Ah yep, you're right about it still being one query regardless :)

Cool, can wait for @Fosome's scope then!